src/IssueViz.UnitTests/Commands/IssueVisualizationToolWindowCommandTests.cs

```diff
@@ -0,0 +1,165 @@
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("commandService");
+
+            act = () => new IssueVisualizationToolWindowCommand(package.Object, commandService.Object, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("logger");
+        }
+
+        [TestMethod]
+        public void Ctor_CommandAddedToMenu()
+        {
+            new IssueVisualizationToolWindowCommand(package.Object, commandService.Object, logger.Object);
+
+            commandService.Verify(x =>
+                    x.AddCommand(It.Is((MenuCommand c) =>
+                        c.CommandID.Guid == IssueVisualizationToolWindowCommand.CommandSet &&
+                        c.CommandID.ID == IssueVisualizationToolWindowCommand.CommandId)),
+                Times.Once);
+        }
+
+        [TestMethod]
+        public void Execute_WindowNotCreated_NoException()
+        {
+            SetupWindowCreation(null);
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("cannot create window frame");
+        }
+
+        [TestMethod]
+        public void Execute_WindowCreatedWithoutFrame_NoException()
+        {
+            SetupWindowCreation(new ToolWindowPane());
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("cannot create window frame");
+        }
+
+        [TestMethod]
+        public void Execute_ExceptionInCreatingWindow_NoException()
+        {
+            SetupWindowCreation(new ToolWindowPane {Frame = Mock.Of<IVsWindowFrame>()}, new NotImplementedException("this is a test"));
+
+            VerifyExecutionDoesNotThrow();
+
+            VerifyMessageLogged("this is a test");
+        }
+
+        [TestMethod]
+        public void Execute_WindowCreatedWithIVsWindowFrame_WindowIsShown()
+        {
+            var frame = new Mock<IVsWindowFrame>();
+            SetupWindowCreation(new ToolWindowPane { Frame = frame.Object });
+
+            VerifyExecutionDoesNotThrow();
+
+            frame.Verify(x=> x.Show(), Times.Once);
+
+            logger.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Execute_WindowCreatedWithIVsWindowFrame_ExceptionInShowingWindow_NoException()
+        {
+            var frame = new Mock<IVsWindowFrame>();
+            frame.Setup(x => x.Show()).Throws(new NotImplementedException("this is a test"));
+
+            SetupWindowCreation(new ToolWindowPane { Frame = frame.Object });
+
+            VerifyExecutionDoesNotThrow();
+
+            frame.Verify(x => x.Show(), Times.Once);
+
+            VerifyMessageLogged("this is a test");
+        }
+
+        private void SetupWindowCreation(WindowPane windowPane, Exception exceptionToThrow = null)
+        {
+            var setup = package.Protected().Setup<WindowPane>("CreateToolWindow", typeof(IssueVisualizationToolWindow), 0);
```

Comment:
In the code we're calling FindToolWindow which is not virtual. I'm relaying on the fact that FindToolWindow calls CreateToolWindow, which _is_ virtual.